docs/bash_script_style_guide.md

```diff
@@ -0,0 +1,100 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Bash Script Style Guide
+
+General guidelines for bash scripts. (May be applied to any shell scripts on compatible parts.)
+
+1. Use 4 spaces indentation. Spare tabs for Makefiles.
+2. Remove whitespaces from the ends of the lines.
+3. Quote variables: `"$VAR"`. Even in cases where it is not absolutely necessary. Comment cases where quotes explicitly can not be used. Disable shellcheck complaint about unquoted variable like in this example:
+    ```
+    # shellcheck disable=SC2086 # $SIGN_SSHOPTS is purposefully unquoted here, it contains several space separated options for ssh
+    if ssh -n -o BatchMode=yes $SIGN_SSHOPTS "$SIGNING_SRV" "${SIGNING_SRV_PATH}/start.sh" sign "-h=$SHA256SUM" > "$SIGNATURE_FILE"; then
+    ...
+    ```
+4. Use curly braces whenever you are concatenating something after a variable: `"${IMGBUILD}/output"`.
+5. Use `[ ]` and `[[ ]]` instead of `test`.
+6. Use `case` instead of `if - elif` structure if possible.
+7. Put space after hash sign in comments: `# Comment`.
+8. Indent comments the same amount as the line being commented.
+9.  Do not add **x** to string comparisons: `[ “x$VAR” = “xBlah” ]`. It is redundant in modern Bash.
```

Comment:
could be a typo; `==` instead of `=`

Reply:
When using single brackets, single equals is the "sh standard", although double equals would also work in bash.
I'd keep using single equals with single brackets and double equals in double brackets.